src/Test.UnitTests.Sarif.WorkItems/SarifWorkItemFilerTests.cs

```diff
@@ -3,17 +3,165 @@
+                .Setup(x => x.ConnectAsync(It.IsAny<Uri>(), It.IsAny<string>()))
+                .Returns(Task.CompletedTask)
+                .Callback<Uri, string>(
+                    (uri, pat) =>
+                    {
+                        // The following data values flow to us via test constants which are
+                        // captured in the default context object that initialized the filer.
+                        pat.Should().Be(NotActuallyASecret);
+
+                        // We configure the filer with a full ADO URI that contains the account and
+                        // project, e.g., https://dev.azure.com/myaccount/myproject. By the time the
+                        // ADO client receives it, however, the project has been stripped off 
+                        // (because it is not required for making the connection).
+                        AzureDevOpsFilingUri.OriginalString.StartsWith(uri.ToString()).Should().BeTrue();                        
+
+                        // Verify that we received the connection request.
+                        connectCalled = true;
+                    });
+
+            // Our GetClientAsync is overridden to provide our low-level ADO mock.
+            vssConnectionMock
+                .Setup(x => x.GetClientAsync())
+                .ReturnsAsync(workItemTrackingHttpClientMock.Object);
+
+            // FIVE. We are required to inject the low level ADO connection wrapper. We are 
+            //       required to do so due to the complexity of creating and initializing
+            //       required objects. MOQ cannot override constructors and (related)
+            //       in general cannot instantiate a type without a parameterless ctor.
+            //       Even when a concrete class can be instantiated, its various properties
+            //       might not be easily stubbed (as for a non-virtual property accessor).
+            //       For these cases, we need to insert an interface in our system, and 
+            //       create wrappers around those concrete types, where the interface is
+            //       directly mapped or otherwise adapted to the concrete type's methods.
+            //       Moq can then simply manufacture a class that implements that interface
+            //       in order to control behaviors.
+            //
+            //       Rather than introducing a type factory or more sophisticated pattern
+            //       of injection, we use the very simple expedient of declaring an internal
+            //       property to hold a mock instance. In production, if that property is null,
+            //       the system instantiates a wrapper around the standard types for use.
+            
+            var adoFilingClient = (AzureDevOpsFilingClient)filer.FilingClient;
+            adoFilingClient.vssConection = vssConnectionMock.Object;
+
+            void action() => filer.FileWorkItems(sarifLog: SimpleLog);
```

Comment:
>action [](start = 17, length = 6)

Why are we declaring a lambda that calls a function, and then calling the lambda, instead of just calling the function? The action is useful when you expect a throw, as in `action.ShouldThrow<ArgumentNullException>()`.